examples/Expr.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE TemplateHaskell, KindSignatures, DeriveFunctor, TypeFamilies #-}
+module Main where
+
+import Data.Functor.Foldable
+import Data.Functor.Foldable.TH
+
+data Expr a
+    = Lit a
+    | Add (Expr a) (Expr a)
+    | Mul (Expr a) (Expr a)
+  deriving (Show)
+
+makeBaseFunctor ''Expr
+
+expr1 :: Expr Int
+expr1 = Add (Lit 2) (Mul (Lit 3) (Lit 4))
+
+main :: IO ()
+main = do
+    let expr2 = ana divCoalg 55 :: Expr Int
+    print $ cata evalAlg expr1
```

Comment:
Thank you for including a test. But right now, it's quite bare-bones—and long as it compiles (and doesn't throw any runtime exceptions), `Cabal` will interpret any run of this program as a success. It'd be nice to hook up a test framework (e.g., `hspec` or `tasty`) and check that `cata evalAlg expr*` actually evaluates to the expected value.